game/scripts/vscripts/Sections/Chapter3/Chapter3.ts

```diff
@@ -111,279 +43,199 @@ const onStart = (complete: () => void) => {
+    const neutralCampPostion = neutralCamp.GetAbsOrigin();
 
     GameRules.SpawnNeutralCreeps();
-    const neutralCamps = Entities.FindAllByClassname(
-        "npc_dota_neutral_spawner"
-    );
-
-    graph = tg.forkAny([
-        tg.trackGoals(getGoals),
-        tg.seq([
-            tg.wait(1),
-            // Show message and explain what the "neutral spawn indicators" on the minimap are.
-            // Kill them after so that we can make our own.
-            tg.immediate(() => DestroyNeutrals()),
-
-
-            tg.immediate(
-                (context) =>
-                    (context[NeutralGoalKeys.MoveToCamp] = GoalState.Started)
-            ),
-            tg.goToLocation(markerLocation),
-            tg.immediate((context) => {
-                context[NeutralGoalKeys.MoveToCamp] = GoalState.Completed;
-                context[NeutralGoalKeys.UpgradeAbility] = GoalState.Started;
-            }),
-
-            // Tell the player to skill dragon blood for extra armor, so that it's better against attacks (big part of fighting neutrals is not taking too much damage)
-            tg.immediate((context) => {
-                if (playerHero.GetLevel() < 2) {
-                    playerHero.HeroLevelUp(false);
+
+    graph = tg.withGoals(_ => goalTracker.getGoals(), tg.seq([
+        tg.wait(1),
+        // Show message and explain what the "neutral spawn indicators" on the minimap are.
+        // Kill them after so that we can make our own.
+        tg.immediate(_ => DestroyNeutrals()),
+
+        tg.immediate(_ => goalMoveToCamp.start()),
+        tg.goToLocation(markerLocation),
+        tg.immediate(_ => {
+            goalMoveToCamp.complete();
+            goalKillSatyrs.start();
+        }),
+
+        // Create the units
+        tg.spawnUnit(
+            "npc_dota_neutral_satyr_trickster",
+            neutralCampPostion.__add(RandomVector(50)),
+            DotaTeam.NEUTRALS,
+            "trickster_1"
+        ),
+        tg.spawnUnit(
+            "npc_dota_neutral_satyr_trickster",
+            neutralCampPostion.__add(RandomVector(50)),
+            DotaTeam.NEUTRALS,
+            "trickster_2"
+        ),
+        tg.spawnUnit(
+            "npc_dota_neutral_satyr_soulstealer",
+            neutralCampPostion.__add(RandomVector(50)),
+            DotaTeam.NEUTRALS,
+            "soulstealer_1"
+        ),
+        tg.spawnUnit(
+            "npc_dota_neutral_satyr_soulstealer",
+            neutralCampPostion.__add(RandomVector(50)),
+            DotaTeam.NEUTRALS,
+            "soulstealer_2"
+        ),
+
+        tg.immediate((ctx) => {
+            units.push(ctx["trickster_1"] as CDOTA_BaseNPC);
+            units.push(ctx["trickster_2"] as CDOTA_BaseNPC);
+            units.push(ctx["soulstealer_1"] as CDOTA_BaseNPC);
+        }),
+
+        // Check if they are killed
+        tg.fork([
+            tg.completeOnCheck(_ => !units.some(unitIsValidAndAlive), 1),
+            //This one gives the item
```

Comment:
Why not the last one that dies?

Reply:
That's what this is or did I misunderstand? It checks that none of the units are valid or alive (ie. that not at least one is valid or alive).

Reply:
This will all be replaced by the ingame function to spawncreeps and the entity_killed event